plugins/reach/lib/model/ReachProfile.php

```diff
@@ -483,12 +483,20 @@ public function fulfillsRules(kScope $scope, $checkEmptyRulesOnly = false)
 			if($gotNonBooleanCondition && $checkEmptyRulesOnly )
 				continue;
 
-			if(!$checkEmptyRulesOnly &&
-				(!count($rule->getConditions()) ||
-					(count($rule->getConditions()) == 1 &&
-						$rule->getConditions()[0]->getType() == ConditionType::BOOLEAN )))
+			if(!$checkEmptyRulesOnly)
 			{
-				continue;
+				if (!count($rule->getConditions()))
+				{
+					continue;
+				}
+				else if (count($rule->getConditions()) == 1)
+				{
+					$condition = $rule->getConditions()[0];
```

Comment:
This wont work in old php 
try - 
php -r "function a() {return array(1,2,33);}print a()[0];"

But you can do this
$condition[0]->getType();